app/src/main/java/com/swent/assos/model/service/impl/DbServiceImpl.kt

```diff
@@ -198,42 +252,98 @@ constructor(
           id = it.id,
           title = it.getString("title") ?: "",
           description = it.getString("description") ?: "",
-          date = it.getString("date") ?: "",
           associationId = it.getString("associationId") ?: "",
           image = it.getString("image") ?: "",
+          startTime = timestampToLocalDateTime(it.getTimestamp("startTime")),
+          endTime = timestampToLocalDateTime(it.getTimestamp("endTime")),
+          fields =
+              if (it.get("fields") is List<*>) {
+                (it.get("fields") as List<*>).filterIsInstance<Map<String, String>>().map { map ->
+                  when (map["type"]) {
+                    EventFieldType.IMAGE.toString() ->
+                        EventFieldImage(title = map["title"] ?: "", image = map["value"] ?: "")
+                    EventFieldType.TEXT.toString() ->
+                        EventFieldText(title = map["title"] ?: "", text = map["value"] ?: "")
+                    else -> EventFieldText("", "")
+                  }
+                }
+              } else {
+                emptyList()
+              },
           documentSnapshot = it)
     }
   }
 
+  override suspend fun createEvent(
+      associationId: String,
+      event: Event,
+      onSuccess: () -> Unit,
+      onError: (String) -> Unit
+  ) {
```

Comment:
The `associationId` is included in the `event`, so it is not necessary to include it as a parameter.